dll/win32/imm32/precomp.h

```diff
@@ -72,7 +72,10 @@ BOOL Imm32GetSystemLibraryPath(LPWSTR pszPath, DWORD cchPath, LPCWSTR pszFileNam
 VOID APIENTRY LogFontAnsiToWide(const LOGFONTA *plfA, LPLOGFONTW plfW);
 VOID APIENTRY LogFontWideToAnsi(const LOGFONTW *plfW, LPLOGFONTA plfA);
 PWND FASTCALL ValidateHwndNoErr(HWND hwnd);
+
 LPVOID APIENTRY Imm32HeapAlloc(DWORD dwFlags, DWORD dwBytes);
+#define Imm32HeapFree(lpData) HeapFree(g_hImm32Heap, 0, (lpData))
```

Comment:
You don't check whether lpData could be NULL? (Code where you did the change usually checks for pointer being not NULL before calling HeapFree).

Reply:
See above @learn-more's comment. NULL check for `HeapFree` can be omitted.

Reply:
https://docs.microsoft.com/en-us/windows/win32/api/heapapi/nf-heapapi-heapfree
![image](https://user-images.githubusercontent.com/2107452/133741759-c8fc16c6-993d-41d2-94ac-21aadcec20b4.png)

Reply:
At the same time there is this:
https://git.reactos.org/?p=reactos.git;a=commit;h=2c48227400a64c07647f1a630c99491a0804cd97
> [CRT]
> - C standard for free says "if pointer is NULL, no action is taken".
> - MSDN for HeapFree says "if pointer is NULL, behavior is undefined".
> Thanks to Thomas for teaching wine guys how to read.

Reply:
The info of MSDN `HeapFree` is misinformation. We can confirm that the NULL pointer is valid:
https://github.com/reactos/reactos/blob/master/sdk/lib/rtl/heap.c#L2282

Reply:
and this is a dependence of an implementation detail.